File: LeetCode/7/reverse.py
class Solution:
    def reverse1(self, x: int) -> int:
        if x == 0:
            return 0

        if x < 0:
            x *= -1
            is_negative = True
        else:
            is_negative = False

        max_32 = 0x80000000 if is_negative else 0x7fffffff

        result = 0
        while x > 0:
            next_digit = x % 10
            next_result = (result * 10) + next_digit
            if next_result > max_32:
                return 0
            else:
                result = (result * 10) + next_digit
                x //= 10

        if is_negative:
            return result*-1
        else:
            return result

File: LeetCode/0007-reverse-integer/reverse.js
var reverse = function(x) {
    if (x > -10 && x < 10) {
        return x
    }

    isNegative = false
    if (x < 0) {
        isNegative = true
        x *= -1
    }

    let max = 0
    if (isNegative) {
        max = 0x80000000
    } else {
        max = 0x7fffffff
    }

    let result = 0
    while (x > 0) {
        let nextStep = x % 10
        let nextResult = result*10+nextStep
        if (nextResult > max) {
            return 0
        } else {
            result = nextResult
            x = Math.floor(x/10)
        }
    }
    if (isNegative) {
        return result*-1
    } else {
        return result
    }
};
